src/github.com/getlantern/flashlight/client/chained.go

```diff
@@ -105,6 +128,117 @@ func (s *ChainedServerInfo) Dialer(deviceID string) (*balancer.Dialer, error) {
 
 			return conn, nil
 		},
+		Check: func() bool {
+			return s.check(d, deviceID)
+		},
 		OnRequest: ccfg.OnRequest,
 	}, nil
 }
+
+func (s *chainedServer) attachHeaders(req *http.Request, deviceID string) {
+	authToken := s.AuthToken
+	if ForceAuthToken != "" {
+		authToken = ForceAuthToken
+	}
+	if authToken != "" {
+		req.Header.Add("X-LANTERN-AUTH-TOKEN", authToken)
+	}
+	req.Header.Set("X-LANTERN-DEVICE-ID", deviceID)
+}
+
+func (s *chainedServer) check(dial func(string, string) (net.Conn, error), deviceID string) bool {
+	rt := &http.Transport{
+		DisableKeepAlives: true,
+		Dial:              dial,
+	}
+	var url string
+	checkTarget := s.checkTargets.get()
+	if checkTarget == "" {
+		url = "http://ping-chained-server"
+	} else {
+		url = fmt.Sprintf("http://%s/index.html", checkTarget)
```

Comment:
You made me think that we probably want to check both HTTP and HTTPS. HTTPS is way more popular while HTTP forwarding is more sophisticated. Maybe not over implement this time.